libraries/AP_NavEKF2/AP_NavEKF2_Measurements.cpp

```diff
@@ -529,6 +529,17 @@ void NavEKF2_core::readGpsData()
 
             }
 
+            if (gpsGoodToAlign && !have_table_earth_field) {
+                table_earth_field_ga = AP_Declination::get_earth_field_ga(gpsloc);
+                table_declination = radians(AP_Declination::get_declination(gpsloc.lat*1.0e-7,
+                                                                            gpsloc.lng*1.0e-7));
+                have_table_earth_field = true;
+                if (frontend->_mag_ef_limit > 0) {
+                    // initialise earth field from tables
+                    stateStruct.earth_magfield = table_earth_field_ga;
```

Comment:
We initialise the earth field to the WMM tables at this point, but in the NavEKF2_core::calcQuatAndFieldStates function the field can subsequently be reset to a value based on the declination, the body frame measurement and the estimated roll and pitch angles. These will then be modified by the ConstrainStates() function so this will probably work OK but the behaviour immediately after these in-flight resets is something we need to monitor during testing.

Otherwise we would need to modify NavEKF2_core::calcQuatAndFieldStates to also do the same table based reset.

Reply:
fixed, thanks

Reply:
fixed